### PR TITLE
Support crd-specific annotations

### PR DESCRIPTION
**Pull request by awalford16**
Adding configuration to have annotations specifically for CRD resources

We are running into this issue: https://github.com/kedacore/keda/issues/4740

We want to set ServerSideApply annotation on our Argo but only for CRD resources and not all resources managed by Keda chart

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] README is updated with new configuration values *(if applicable)* [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#documentation)
- [x] A PR is opened to update KEDA core ([repo](https://github.com/kedacore/keda)) *(if applicable, ie. when deployment manifests are modified)*

Fixes #

### CODE REVIEW

File: keda/Chart.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ kubeVersion: ">=v1.23.0-0"
 
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
-version: 2.12.1
+version: 2.12.2
```
Comment (tomkerkhove) [inline keda/Chart.yaml:11]: Please revert this change

File: keda/README.md [hunk 2]
```diff
@@ -72,6 +72,7 @@ their default values.
 | `clusterDomain` | string | `"cluster.local"` | Kubernetes cluster domain |
 | `clusterName` | string | `"kubernetes-default"` | Kubernetes cluster name. Used in features such as emitting CloudEvents |
 | `crds.install` | bool | `true` | Defines whether the KEDA CRDs have to be installed or not. |
+| `crds.additionalAnnotations` | object | `{}` | Custom annotations to add specifically to CRD resources |
```
Comment (tomkerkhove) [inline keda/README.md:75]: Documentation is generated, can you document this in the values file please?

See https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#documentation

Comment (awalford16) [reply keda/README.md]: Updated

**Comment by tomkerkhove** [comment]
Can you fix the DCO issue please?

**Comment by tomkerkhove** [comment]
@JorTurFer Any objections for merging? I personally think we should be good.

**Comment by JorTurFer** [comment]
Thanks for the improvement!
